csharp/extractor/Semmle.Extraction.CSharp/Entities/Method.cs

```diff
@@ -243,7 +243,12 @@ public static void AddExplicitInterfaceQualifierToId(Context cx, EscapingTextWri
             if (methodKind == MethodKind.ExplicitInterfaceImplementation)
             {
                 // Retrieve the original method kind
-                methodKind = methodDecl.ExplicitInterfaceImplementations.Select(m => m.MethodKind).FirstOrDefault();
+                if (methodDecl.ExplicitInterfaceImplementations.IsEmpty)
+                {
+                    throw new InternalError(methodDecl, $"Couldn't get the original method kind for explicit interface implementation");
```

Comment:
Question: I see that the `methodDecl` is given to `InternalError` as its first argument. To what extent will information about `methodDecl` be visible in logs? The `Message` implementation in `InternalError` only uses the error text. There are a few places in the codebase which catch `InternalError`s and may access the `EntityText` / `Location` properties, but it's not clear to me whether that happens for `InternalError`s thrown here.

Also, small wording improvement:
```suggestion
                    throw new InternalError(methodDecl, $"Couldn't get the original method kind for an explicit interface implementation");
```